stemcell_builder/lib/prelude_fips.bash

```diff
@@ -0,0 +1,141 @@
+# do nothing if the variant isn't fips
+if [ ${stemcell_operating_system_variant} != "fips" ]; then
+    echo "Skipping base_apt_fips given that this is not a 'fips' variant"
+    exit 0
+else
+    echo "FIPS variant detected. setting up FIPS"
+fi
+
+if [ -z ${UBUNTU_ADVANTAGE_TOKEN} ]; then
+    echo "'fips' variant detected but \$UBUNTU_ADVANTAGE_TOKEN not given."
```

Comment:
It seems like we are conflating `UBUNTU_ADVANTAGE_TOKEN` and "FIPS variant" here. The existence of `UBUNTU_ADVANTAGE_TOKEN` is a (possibly required) implementation detail of assembling a FIPS variant. 

I would suggest that we use an explicit "FIPS" variable here since an `UBUNTU_ADVANTAGE_TOKEN` could also be used in cases when a stemcell is being built with Extended Support packages.